base/utils/R/convert.input.R

```diff
@@ -618,6 +618,18 @@ convert.input <- function(input.id, outfolder, formatname, mimetype, site.id, st
   }
   
   #--------------------------------------------------------------------------------------------------#
+  # Check if result has empty or missing files
+  
+  result_sizes <- result %>% purrr::map(~ .x$file) %>% purrr::map(~ .x %>% file.size) %>% as.data.frame() %>% 
+    `colnames<-`(c("file_size")) %>%
+    mutate(missing = ifelse(is.na(file_size),NA,TRUE)) %>%
+    mutate( empty = ifelse(file_size==0,"EMPTY",TRUE))
```

Comment:
These should be `dplyr::mutate`. Also, `mutate` evaluates sequentially, so you can combine these into one statement. I would write this as:

```
result %>%
  dplyr::mutate(
    file_size = purrr::map_dbl(file, file.size),
    missing = is.na(file_size),
    empty = file_size == 0
)
```

Explanation:
- `dplyr::if_else` is better than `base::ifelse` because it does stricter type checking. `ifelse` does some implicit coercion that can give unpredictable results.
- Logical values are easier to work with than custom values. Leaving `missing` and `empty` as logical will make it easy later on to do things like `dplyr::if_else(result$missing, val_if_true, val_if_false)` or `if (all(result$missing))`
- Note that in your `empty` column, because a column can only have a single type, `TRUE` will be coerced to the string `"TRUE"` if at least one file is `"EMPTY"`, due to R's coercion rules. You almost definitely want to avoid this.